src/verify.rs

```diff
@@ -378,59 +359,52 @@ fn verify_proof<'a>(state: &mut VerifyState<'a>, stmt: StatementRef<'a>) -> Opti
             for &ch in chunk {
                 if ch >= b'A' && ch <= b'T' {
                     k = k * 20 + (ch - b'A') as usize;
-                    if let Some(err) = execute_step(state, k) {
-                        return Some(err);
-                    }
+                    try!(execute_step(state, k));
                     k = 0;
                     can_save = true;
                 } else if ch >= b'U' && ch <= b'Y' {
                     k = k * 5 + 1 + (ch - b'U') as usize;
                     if k >= (u32::max_value() as usize / 20) - 1 {
-                        return Some(Diagnostic::ProofMalformedVarint);
+                        return Err(Diagnostic::ProofMalformedVarint);
                     }
                     can_save = false;
                 } else if ch == b'Z' {
                     if !can_save {
-                        return Some(Diagnostic::ProofInvalidSave);
+                        return Err(Diagnostic::ProofInvalidSave);
                     }
                     save_step(state);
                     can_save = false;
                 } else if ch == b'?' {
-                    if k > 0 {
-                        return Some(Diagnostic::ProofMalformedVarint);
-                    }
-                    return Some(Diagnostic::ProofIncomplete);
+                    return Err(if k > 0 {
```

Comment:
These are two unrelated error conditions, it seems strange to combine them as a single error return